scipy/stats/_continuous_distns.py

```diff
@@ -5909,19 +5914,33 @@ def _isf(self, q, df):
         return -sc.stdtrit(df, q)
 
     def _stats(self, df):
-        mu = np.where(df > 1, 0.0, np.inf)
-        mu2 = _lazywhere(df > 2, (df,),
-                         lambda df: df / (df-2.0),
-                         np.inf)
-        mu2 = np.where(df <= 1, np.nan, mu2)
-        g1 = np.where(df > 3, 0.0, np.nan)
-        g2 = _lazywhere(df > 4, (df,),
-                        lambda df: 6.0 / (df-4.0),
-                        np.inf)
-        g2 = np.where(df <= 2, np.nan, g2)
+        # 0.0, 1.0, 0.0, 0.0
+        cond = df == np.inf
+        mu = np.where(df > 1 | cond, 0.0, np.inf)
+
+        def _mu2_finitedf(df):
+            mu2 = _lazywhere(df > 2, (df,),
+                             lambda df: df / (df-2.0),
+                             np.inf)
+            mu2 = np.where(df <= 1, np.nan, mu2)
+            return mu2
+
+        mu2 = _lazywhere(~cond, (df,), _mu2_finitedf, 1.0)
+        g1 = np.where(df > 3 | cond, 0.0, np.nan)
+
+        def _g2_finitedf(df):
+            g2 = _lazywhere(df > 4, (df,),
+                            lambda df: 6.0 / (df-4.0),
+                            np.inf)
+            g2 = np.where(df <= 2, np.nan, g2)
+            return g2
+
+        g2 = _lazywhere(~cond, (df,), _g2_finitedf, 0.0)
         return mu, mu2, g1, g2
 
     def _entropy(self, df):
+        if df == np.inf:
+            return norm._entropy()
```

Comment:
Note to self... when we rewrite `rv_continuous`, have a consistent policy for what is passed into these methods. I was not aware that `_entropy` would be automatically vectorized.

That said, it would be nice if the SciPy developer had the choice to implement either a vectorized `_<methodname>_vectorized` or an `_<methodname>_scalar`.  To save future headaches, `<methodname>` should do the following before calling `_<methodname>_vectorized`:

- Ensure that all the inputs are at least 1d arrays
- _Broadcast all the input arrays together_
- Take care of returning a scalar or 0d array instead of a 1d array as needed, depending on what the user passed into `<methodname>`